mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,167 @@
+        self.get_filesystem(silent=False)
+
+    def get_filesystem(self, silent=True):
+        """return fsspec file system object, if supported"""
+        if not self._filesystem:
+            self._filesystem = fsspec.filesystem("dbfs", **self.get_storage_options())
+        return self._filesystem
+
+    def get_storage_options(self):
+        return dict(
+            token=self._get_secret_or_env("DATABRICKS_TOKEN"), instance=self.endpoint
+        )
+
+    def _verify_filesystem_and_key(self, key: str):
+        if not self._filesystem:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Performing actions on data-item without a valid filesystem"
+            )
+        if not key.startswith("/"):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid key parameter - key must start with '/'"
+            )
+
+    def get(self, key: str, size=None, offset=0) -> bytes:
+        self._verify_filesystem_and_key(key)
+        if size is not None and size <= 0:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "size cannot be negative or zero"
+            )
+        if not size and not offset:
+            return self._filesystem.cat_file(key)
+        end = offset + size if size is not None else None
+        return self._filesystem.cat_file(key, start=offset, end=end)
+
+    def put(self, key, data, append=False):
+
+        self._verify_filesystem_and_key(key)
+        if append:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Append mode not supported for Databricks file system"
+            )
+        #  can not use append mode because it overrides data.
+        mode = "w"
+        if isinstance(data, bytes):
+            mode += "b"
+        elif not isinstance(data, str):
+            raise TypeError(f"Unknown data type {type(data)}")
+        with self._filesystem.open(key, mode) as f:
+            f.write(data)
+
+    def upload(self, key: str, src_path: str):
+        self._verify_filesystem_and_key(key)
+        self._filesystem.put_file(src_path, key, overwrite=True)
+
+    def stat(self, key: str):
+        self._verify_filesystem_and_key(key)
+        file = self._filesystem.stat(key)
+        if file["type"] == "file":
+            size = file["size"]
+        elif file["type"] == "directory":
+            raise FileNotFoundError("Operation expects a file not a directory!")
+        return FileStats(size, None)
+
+    def listdir(self, key: str):
+        """
+        Basic ls of file/dir - without recursion.
+        """
+        self._verify_filesystem_and_key(key)
+        if self._filesystem.isfile(key):
+            return key
+        remote_path = f"{key}/*"
+        files = self._filesystem.glob(remote_path)
+        #  Get only the files and directories under key path, without the key path itself.
+        # for example for /test_mlrun_dbfs_objects/test.txt the function will return ['test.txt'].
```

Comment:
It's a bit confusing as an example, because `test.txt` is not under `/test_mlrun_dbfs_objects/test.txt`. It's the file name part of the path, but refers to the same file.